sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -134,4 +136,24 @@ export class ApplicationController extends BaseController {
 
     return assessment;
   }
+
+  /**
+   * Confirm Assessment of a Student.
+   * @param applicationId application id to be updated.
+   */
+  @AllowAuthorizedParty(AuthorizedParties.student)
+  @Patch(":applicationId/confirm-assessment")
+  async studentConfirmAssessment(
+    @Param("applicationId") applicationId: number,
+  ): Promise<any> {
+    const updateResult = await this.applicationService.studentConfirmAssessment(
+      applicationId,
+    );
+    if (!updateResult) {
+      throw new UnprocessableEntityException(
+        `Confirmation of Assessment for the application id ${applicationId} failed.`,
+      );
+    }
+    return updateResult;
```

Comment:
Is there a reason why we are returning the updateResult here?
If the method is executed with success it should return a proper HTTP status, if there is an error the proper HTTP status should also be returned.